### DISCUSSION TITLE
B1 Energy export misreported

### DISCUSSION THREAD

**Discussion by Mougli** [WillCodeForCats/solaredge-modbus-multi#657] [General]
### Describe the bug

In the last two days, the Solaredge B1 Energy export sensor stopped reporting energy consumption correctly.

When running on battery alone, the system reports less than 10 Wh consumed overnight, when other energy monitors I have tell me this should be at least 1.4 kWh.

The SolarEdge monitoring app and websites display consumption numbers in line with my expectations.

Power cycling the inverter did not solve the problem.

### Expected behavior

Data should match what's in the SolarEdge app.

### Screenshots

Screenshot of erroneous energy display in the Energy dashboard, using B1 Energy Export and Import sensors for battery consumption

![image](https://github.com/user-attachments/assets/cbdbd94d-3ae9-456b-8961-a5129e3aec92)

Below are individual device readings for the same period to show the discrepancy

![image](https://github.com/user-attachments/assets/d457ad85-b46f-42c4-ba58-60bda0f9d0ba)

And here's the sensor history over the same period

![image](https://github.com/user-attachments/assets/89a29ea6-eea7-4ede-8ff8-0027603f5e1b)

And the battery charge state for comparison

![image](https://github.com/user-attachments/assets/1452ac7d-4c05-4a64-a321-378ccd1d69d7)

And finally the battery consumption graph from the SolarEdge app for the same period

![image](https://github.com/user-attachments/assets/237586d7-2f53-4b90-ad2b-36eaa89a105b)

### Diagnostic File

[config_entry-solaredge_modbus_multi-bb1cb013aefa0574b96da664b866b169.json](https://github.com/user-attachments/files/16682776/config_entry-solaredge_modbus_multi-bb1cb013aefa0574b96da664b866b169.json)

### Home Assistant Version

2024.8.2

### solaredge-modbus-multi Version

2.4.20

### Installation Type

HAOS

### Read the Instructions

- [X] My configuration follows https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration
- [X] I have read the Known Issues https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues
- [X] This issue is not about a template error (do not report template errors, read https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Template-Design-Notes)

**Comment by WillCodeForCats** [comment]
I don't have any control over what data the inverter reports over modbus. Since I don't have batteries personally maybe someone else has an idea.

**Comment by bakboord4** [comment]
@Mougli To my opinion it is difficult to pinpoint a misreporting in HA Energy dashboard to the Solaredge-modbus-mult. 

HA Energy dashboard is showing per hour accumulated values. 
You argue 'at least' 1.4 kWh. Is this a per hour accumulated value or a value at a given moment?
Is it possible to point in the above presented charts, your claim of 1.4 kWh, sensor name included?

If your wish is that 'Data should match what's in the SolarEdge app', my suggestion is that additional information is needed about what exactly is misreported. 
What value it should be, sensor name included (battery and/or solaredge-multi-bus) and 
what exactly is presented in the Energy-dasboard including all the used sensors names (with yaml) in between and possible automations used within HA.

**Reply by Mougli** [reply]
I don't know what more information I can give you.

It should be pretty clear from the screenshot that when a few device draw a constant 200-300W (or 200-300Wh every hour), the whole system shouldn't report 0 Wh consumed.

It should also be clear that the sensor history shouldn't flatline when there is a constant power draw on the battery.

This was working fine for months, just started acting up a couple days ago, and I have made no changes.

**Reply by WillCodeForCats** [reply]
The integration doesn't alter data. It simply returns what the inverter gives, maybe applies a scale factor or units (unit settings are in the HA frontend) and does some sanity checks to try and avoid displaying bad data.

You can turn on debug logging and analyze how the value changes. Look for the value of `B_Export_Energy_WH`. The native value for battery energy is watt-hours.

Others have reported in the past they need to power cycle their entire system to re-sync. This is documented in Known Issues: https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues

**Reply by Mougli** [reply]
I mentioned in the report that power cycling did not resolve the issue.

I understand based on your comments that the issue is likely not on your side, but I hope you can understand my confusion when the inverter seems to report one value through its app and something completely different via modbus.

I'll contact SolarEdge and report back any findings here in case it can help someone else.

**Reply by WillCodeForCats** [reply]
Did any firmware versions change recently?

**Reply by Mougli** [reply]
Not that I'm aware of, version number sensor reports the same value throughout.

I contacted SolarEdge support, after I explained the issue they remotely removed/re-added the energy meter to the system, and things seem to back to normal now, but I'll keep an eye on it.

**Reply by Mougli** [reply]
aaand nope, it's flatlining again. I'll get back to them.

**Comment by Mougli** [comment]
Still discussing with SolarEdge as the issue is not resolved, but thought I'd share some additional information in case someone else encounters the same symptoms.

It seems that there's a consumption threshold (which seems to be somewhere between 300 and 500 Watts) below which the battery will seemingly decide not to report any increase in energy exported out of it ( just to give the example again, with a constant(ish) 250W power draw, I should expect to see an increase of 0.25 kWh over the course of an hour, but the sensor reads an increase of 0.017 kWh over the course of 7 hours.)
